### PR TITLE
Add HGDP + 1KG subset option to filter table creation

### PR DESCRIPTION
**Pull request by jkgoodrich**
I realized when looking over the HGDP + 1KG sanity checks that we don't have variant filtering information (NA below) for variants strictly found in samples that didn’t pass gnomAD QC (including the outlier sample QC). This is because of this step: https://github.com/broadinstitute/gnomad_qc/blob/8daf7b1c974dada6f306f2a8f65eeb8e5b007046/gnomad_qc/v3/variant_qc/evaluation.py#L62 where we filter the VQSR HT to only variants found in “high quality samples”. So, we do have VQSR results for these variants and the changes to this code add an HGDP + 1KG option where the variants are not filtered.

```
+-------------+-----------+-----------------+--------------------+----------+ 3]
| is_filtered |         n | frac_any_filter | frac_inbreed_coeff | frac_ac0 |
+-------------+-----------+-----------------+--------------------+----------+
|        bool |     int64 |         float64 |            float64 |  float64 |
+-------------+-----------+-----------------+--------------------+----------+
|       False | 157929744 |        0.00e+00 |           0.00e+00 | 0.00e+00 |
|        True |  27265739 |        1.00e+00 |           9.99e-03 | 4.09e-01 |
|          NA |   4186478 |        0.00e+00 |           0.00e+00 | 0.00e+00 |
+-------------+-----------+-----------------+--------------------+----------+
```

Includes changes to the resources so we don't overwrite the filtering HTs used for the full subset and adds allele data onto the filtering HT because that annotation is added with `**rf_ht[ht.key]` and that has already been filtered to only variants found in high quality samples and we use this annotation for the HGDP + 1KG variant annotation HT. I also modified the expressions for the AC0 and monoallelic filters.

### CODE REVIEW

File: gnomad_qc/v3/variant_qc/final_filter.py [hunk 1]
```diff
@@ -372,6 +400,11 @@ def main(args):
         choices=["vqsr_classic", "vqsr_alleleSpecific", "vqsr_alleleSpecificTrans"],
         type=str,
     )
+    parser.add_argument(
+        "--hgdp_1kg_subset",
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/variant_qc/final_filter.py:404]: I will switch to `-` in the future, but all of v2 and v3 currently have `_` so not worth it to change them all now. Will keep as is for consistency.

File: gnomad_qc/v3/variant_qc/final_filter.py [hunk 2]
```diff
@@ -241,9 +245,33 @@ def main(args):
...
-    freq_ht = get_freq().ht()
+    if CURRENT_RELEASE == "3.1.2":
+        freq_ht = release_sites().versions["3.1.2"].ht()
+        freq_ht = freq_ht.select("freq", InbreedingCoeff=freq_ht.info.InbreedingCoeff)
+    else:
+        freq_ht = get_freq().ht()
+
     ht = ht.annotate(InbreedingCoeff=freq_ht[ht.key].InbreedingCoeff)
     freq_idx = freq_ht[ht.key]
+    if args.hgdp_1kg_subset:
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/variant_qc/final_filter.py:256]: To be consistent I will change the code base to have tgp (and therefore be consistent with what is in gnomad_methods). When we refer to it in blog posts, notebooks, and on the downloads page we use 1KG

File: gnomad_qc/v3/variant_qc/final_filter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/variant_qc/final_filter.py:404]: I will also keep the use of a verb for store_true args in mind in the future, I think for this I will keep the same flag as I have used in the other scripts that also create files specific to this subset

**Comment by jkgoodrich** [comment]
Can you take a close look over this section? I want to make sure the way I am thinking about the AC0 filter makes sense
```
    if args.hgdp_tgp_subset:
        hgdp_tgp_freq_ht = get_freq(subset="hgdp-tgp").ht()
        hgdp_tgp_freq_idx = hgdp_tgp_freq_ht[ht.key]
        # If this is AC0 in full gnomAD and AC0 in HGDP + 1KG/TGP mark it as AC0
        # If missing in full gnomAD only check HGDP + 1KG/TGP,
        # If missing in HGDP + 1KG/TGP only check full gnomAD
        ac0_filter_expr = hl.coalesce(
            (freq_idx.freq[0].AC == 0) & (hgdp_tgp_freq_idx.freq[0].AC == 0),
            hgdp_tgp_freq_idx.freq[0].AC == 0,
            freq_idx.freq[0].AC == 0,
        )
```

**Review by klaricch — COMMENTED** [review]
I think the AC0 logic makes sense

File: gnomad_qc/v3/README.md [hunk 3]
```diff
@@ -62,8 +62,8 @@ The scripts below are run approximately in the order they are listed. We begin q
   * `--export_transmitted_singletons_vcf` - Export transmitted singletons (`--generate_fam_stats`) to VCF files, used for running VQSR with transmitted singletons in the training.
 
 **create_release/**
-* `create_release_sites_ht.py` - Combine frequency, filtering allele frequency, variant QC, VEP, dbSNP, in silico scores, and variant QC metric histogram annotations into unified table. Add frequency annotations for each sample subset (e.g., controls/biobanks, non-neuro, non-cancer, non-TOPMed, non-v2, 1KG, HGDP).
-* `create_hgdp_tgp_subset.py` - Subset raw sparse MatrixTable to only samples in the Human Genome Diversity Project (HGDP) and 1000 Genomes Project (1KG/TGP) to create an unsplit sparse MatrixTable and split dense MatrixTable (with full sample and variant annotations) that can be made publicly available.
+* `create_release_sites_ht.py` - Combine frequency, filtering allele frequency, variant QC, VEP, dbSNP, in silico scores, and variant QC metric histogram annotations into unified table. Add frequency annotations for each sample subset (e.g., controls/biobanks, non-neuro, non-cancer, non-TOPMed, non-v2, CURRENT_VERSION, HGDP).
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/README.md:65]: ugh, no thank you for catching! I did find and replace for 1kg, but the replace had the wrong thing at first, I thought I undid all the ones that were accidentally changed

**Review by klaricch — APPROVED** [review]

**Pull request merged**